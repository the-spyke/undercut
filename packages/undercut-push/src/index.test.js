import { describe, expect, test } from "@jest/globals";

import * as exports from "./index.js";

import {
	push,
	filter, map, skip, sum, take,
	toArray,
} from "./index.js";

describe(`Push`, () => {
	test(`exports of the main "push" entry should be stable`, () => {
		expect(Object.keys(exports).sort()).toMatchSnapshot();
	});

	test(`simple push scenarios`, () => {
		const data1 = [1, 2, 3, 4, 5, 6, 7];
		const pipeline1 = [
			map(x => x + 3),
			skip(1),
			take(100),
			map(x => x - 3),
			filter(x => x !== 4),
		];

		expect(
			push(toArray(), pipeline1, data1).values
		).toEqual([2, 3, 5, 6, 7]);

		expect(
			push(toArray(), [
				...pipeline1,
				sum(),
				map(x => x + 0.5),
			], data1).values
		).toEqual([23.5]);
	});
});
