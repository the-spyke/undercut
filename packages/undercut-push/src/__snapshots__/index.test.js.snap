// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Push exports of the main "push" entry should be stable 1`] = `
Array [
  "append",
  "average",
  "buffer",
  "bufferAll",
  "chunk",
  "collect",
  "collectArray",
  "collectMap",
  "collectObject",
  "collectSet",
  "compact",
  "composeOperations",
  "concatEnd",
  "concatStart",
  "count",
  "difference",
  "differenceBy",
  "every",
  "filter",
  "find",
  "findIndex",
  "first",
  "flatMap",
  "flatten",
  "flattenArrays",
  "flattenIterables",
  "forEach",
  "groupBy",
  "includes",
  "interleave",
  "intersection",
  "intersectionBy",
  "join",
  "last",
  "map",
  "max",
  "min",
  "nth",
  "orderBy",
  "prepend",
  "push",
  "pushArray",
  "pushLine",
  "pushValue",
  "reduce",
  "remove",
  "reverse",
  "skip",
  "skipWhile",
  "some",
  "sort",
  "sortNumbers",
  "sortStrings",
  "sum",
  "symmetricDifference",
  "symmetricDifferenceBy",
  "take",
  "takeWhile",
  "toArray",
  "toConsumer",
  "toNull",
  "toValue",
  "union",
  "unionBy",
  "unique",
  "uniqueBy",
  "unzip",
  "unzipWith",
  "zip",
  "zipWith",
]
`;
